packages/frontend/src/components/FlowStepTestController/TestResult.tsx

```diff
@@ -84,9 +90,13 @@ export default function TestResult(props: TestResultsProps): JSX.Element {
 
     return (
       <Box w="100%">
-        {isMock && (
+        {(isMock || isForEachStep) && (
           <Infobox variant="info">
-            <Text>{getMockDataMessage(selectedActionOrTrigger)}</Text>
+            <Text>
+              {isForEachStep
+                ? getForEachDataMessage(testExecutionSteps, step)
+                : getMockDataMessage(selectedActionOrTrigger)}
+            </Text>
```

Comment:
TODO: we should refactor this actually, make it all client side. mockDataMsg only serves formsg mock data so it's quite meaningless to pull from backend.